test/FunctionalTests/Microsoft.OData.Core.Tests/Json/JsonValueUtilsTests.cs

```diff
@@ -98,11 +98,71 @@ public void WriteMultipleSpecialCharactersShouldWork()
             foreach (string specialChar in this.escapedCharMap.Keys)
             {
                 this.TestInit();
-                JsonValueUtils.WriteEscapedJsonString(this.writer, string.Format("{0}Start{0}Middle{0}End", specialChar));
+                JsonValueUtils.WriteEscapedJsonString(this.writer, string.Format("{0}Start{0}Middle{0}End", specialChar), ref this.buffer);
                 this.StreamToString().Should().Be(string.Format("\"{0}Start{0}Middle{0}End\"", this.escapedCharMap[specialChar]));
             }
         }
 
+        [Fact]
+        public void WriteMultipleSpecialCharactersAtStartOfBufferLengthShouldWork()
+        {
+            foreach (string specialChar in this.escapedCharMap.Keys)
+            {
+                this.TestInit();
+                char[] charBuffer = new char[10];
+                JsonValueUtils.WriteEscapedJsonString(this.writer, string.Format("StartMiddle{0}End", specialChar), ref charBuffer);

```

Comment:
It looks like this test case has only one special character. Did you intend for it to have multiple special characters at the start of the string?

Reply:
Resolved